lib/cli.js

```diff
@@ -0,0 +1,201 @@
+        -k, --kms               a KMS key ID for parameter encryption or
+                                configuration encryption at rest on S3. If not
+                                provided, no encryption will be performed. If
+                                provided as a flag without a value, the default
+                                key id alias/cloudformation will be used.
+        -f, --force             perform a create/update/delete command without any
+                                prompting, accepting all defaults
+        -e, --extended          display resource details with the info command
+        -d, --decrypt           decrypt secure parameters with the info command
+        -p, --parameters        JSON string of parameters to override on create/update
+        -a, --all               Toboggan deploy of all functions found in subdirectories
+    `
+  }, {
+    alias: {
+      c: 'config-bucket',
+      t: 'template-bucket',
+      n: 'name',
+      r: 'region',
+      k: 'kms',
+      f: 'force',
+      e: 'extended',
+      d: 'decrypt',
+      p: 'parameters',
+      a: 'all'
+    },
+    string: ['config-bucket', 'template-bucket', 'name', 'region', 'parameters'],
+    boolean: ['force', 'extended', 'decrypt', 'all'],
+    default: {
+      region: 'us-east-1',
+      force: false,
+      kms: false,
+      extended: false,
+      decrypt: false,
+      parameters: undefined,
+      all: false
+    }
+  });
+
+  // if (cli.flags.true) {
+  //   if (create) {
+  //     // find all functions that don't have a matching stack
+  //   } else {
+  //     // find all stacks that do have a matching stack
+  //   }
+  // }
+
+
+  // check the caller location and set sane template default location
+  if (!cli.flags.all) { // no tobogganing
+    if (!cli.input[2] && cli.input[0] != 'init') {
+      var files = fs.readdirSync(process.cwd());
+      if (files.indexOf('function.template.js') === -1) {
+        throw new Error('function.template.js file not found in ' + process.cwd());
+      }
+      cli.input[2] = './function.template.js';
+      // generate default function name of repoDir-functionDir
+      // stack name then is later set to repoDir-functionDir-environment
+      if (!cli.flags.name) {
+        cli.flags.name = cli.flags.n = (process.cwd().split(path.sep).slice(-2)).join('-');
+      }
+    }
+  }
+
+  if (/init/.test(cli.input[0])) {
+    if (!cli.input[1])
+      throw new Error('Please provide a function name to initialize');
+    if (cli.flags.all)
+      throw new Error('init tobogganing not currently supported');
+  }
+
+  // Make sure that we can set a template bucket for create and update commands
+  if (/create|update/.test(cli.input[0])) {
+    if (!cli.flags.templateBucket && !env.AWS_ACCOUNT_ID)
+      throw new Error('Provide $AWS_ACCOUNT_ID as an environment variable to use the default template bucket, or set --template-bucket');
```

Comment:
Minor grammatical note for the error message - we don't need the comma before `or` here in "or set --template-bucket". Same for line 123.